data/libs/prelude.icicle

```diff
@@ -18,3 +60,20 @@ oldest v = let fold1 s = v : s ~> s.
 
 # Boolean statements
 not a = case a | True -> False | False -> True end.
+
+# Version of num flips which should fail upon finding a tombstone value.
+numflips v = let fold1 s = (v,0) : case s | (s0,i) -> (case (s0 == v) | True -> (v,i) | False -> (v,i+1) end) end ~> case s | (s0,i) -> i end.
+
+# Version of num flips which will accommodate tombstone values. Must have the tombstone feature passed to it.
+numflips' v t
```

Comment:
Yeah, so I was discussing with Amos a better way to test these functions. Because I don't think the current way is very good.
Maybe a test which runs arbitrary data through R and icicle?